maven/lib/dependabot/maven/file_parser/repositories_finder.rb

```diff
@@ -155,6 +156,12 @@ def remote_pom_url(group_id, artifact_id, version, base_repo_url)
             "#{artifact_id}-#{version}.pom"
         end
 
+        def urls_from_credentials
+          @credentials.
+            select { |cred| cred["type"] == "maven_repository" }.
```

Comment:
I had been thinking we'd only do this for creds with `replaces-base: true`. I can see now that you might want this but still allow a public registry though. Is there any potential for breaking existing jobs if we start to automatically move the registry sources to the front?

Alternatives might be:
1. Introducing a new "global" key on the registry to indicate it comes from a global configuration not checked into the repo so we should insert it
2. Use `replaces-base` which would also remove automatic usage of the default registry but allow the default registry to added back via inclusion in `dependabot.yml` registries.

But if it doesn't seem likely this would break anything we could try this out first.

Reply:
My thought was `replaces-base` would remove the default registry if it is true in any of the maven creds. I was going to follow this up by adding that in the next bite-sized PR. 

This approach does allow fallback to Maven Central. For users of GitHub Registry that's probably what they want. It's also possible GHR users would want to check Central first to avoid 404ing first. If we drop the username and password requirement, then they have complete control of the order by specifying in dependabot.yml.